tripy/tripy/backend/mlir/utils.py

```diff
@@ -433,8 +434,16 @@ def get_flat_ir_operation(output_names):
             ]
         )
 
+    # Capture the full error message in case exception is rethrown.
+    full_error_info = traceback.format_exc()
+    new_message = f"{str(exc)}\n\nAdditional context:\n{full_error_info}"
+
+    # Construct the new exception with the formatted message
+    new_exc = type(exc)(new_message)
```

Comment:
Why not just include the traceback in the details field of `raise_error`?

Reply:
It might be better to separate out the details from exceptions vs extra details.